File: Python/00_get-path.py
# -*- coding: utf-8 -*-
"""
Created on Tue Dec  1 19:13:07 2020

@author: david
"""

import os

file_path = os.path.dirname(os.path.realpath(__file__))
file_path = [file_path][0]
os.chdir(file_path)


path_parent = os.path.dirname(os.getcwd())
os.chdir(path_parent)


for sub_dir_name in os.listdir():
    if os.path.isdir(sub_dir_name) and sub_dir_name != '.git':
        str_1 = f'{sub_dir_name}_dir = '
        str_2 = f'"{path_parent}/{sub_dir_name}/"'
        str_2 = str_2.replace('\\','/')
        my_code = str_1 + str_2
        exec(my_code,globals())

    

File: R/00_get-path.R


# Script to define variables for directory names

path <- dirname(rstudioapi::getActiveDocumentContext()$path)
path <- dirname(path)


dir_list <- list.dirs(path,recursive = FALSE)
dir_list <- gsub(paste0(path,"/"), "", dir_list)


for (sub_dir_name in dir_list){
  
    if (sub_dir_name != ".git"){
      
      
      str_1 = paste0(sub_dir_name,"_dir <- ")
      str_2 = paste0("'",path,"/",sub_dir_name,"/'")
      my_code <- paste0(str_1,str_2)
      eval(parse(text=my_code))
      
      
                                }
    
  
  
                               }

